include/onnxruntime/core/optimizer/graph_transformer.h

```diff
@@ -9,15 +9,33 @@
 
 namespace onnxruntime {
 
+/**
+* Default_Global refers to all the transformations which are required for 
+  corrcetness and are execution provider independent
+* Default_ProvderSpecific refers to transformations which are also required 
+  for correctness but are execution provider specific
+* Optional_L1 refers to all basic optmizations. These are run both before and 
+  after graph partitioning
+* Optional_L2 refers to all fusion optmizations. These are run after graph 
+  partitioning
+*/
+enum TransformerLevel {
+  Default_Global = 0,
+  Default_ProviderSpecific = 1,
+  Optional_L1 = 2,
+  Optional_L2 = 4,  
+  MaxTransformerLevel = 8
+};
+
 /**
 @class GraphTransformer
 
 The interface for in-place transformation of a Graph.
 */
 class GraphTransformer {
  public:
-  GraphTransformer(const std::string& name, const std::string& desc)
-      : name_(name), desc_(desc) {
+  GraphTransformer(const std::string& name, const std::string& desc, const TransformerLevel level, std::vector<std::string>&& providers)
```

Comment:
>std::vector<std::string>&& providers [](start = 99, length = 36)

You could default providers to {} 

Is the expectation that this list is empty unless it's a provider specific transformer? Just trying to understand the usage when there are some fairly generic things (eg. conv_add_fusion) that seem like they're not provider specific, but you're passing in just the CPUExecutionProvider as a valid value. If I add a new execution provider type, am I expected to go find every single applicable transformer and update the list? 

I think it's probably cleaner and more scalable for the ExecutionProvider to either give you the list of transformers it wants to register (you'll need to deal with multiple providers registering things with the same name), or for you to ask the execution provider to apply its transforms. Either way, the execution provider should determine which transformers it can use, rather than the transformer having a list of supported execution providers.

Reply:
I agree that we may have default values for the two fields. maybe designing "setter" for them is an idea, so that it could be set with specific data, otherwise default value is used. Like OperatorSchema/Kernel registration.